src/hooks/usePWAInstall.ts

```diff
@@ -18,15 +20,54 @@ export function usePWAInstall() {
     const [isInstallable, setIsInstallable] = useState(false);
     const [isPrompting, setIsPrompting] = useState(false);
 
-    // Check if user previously dismissed auto-prompts (persisted in localStorage)
+    // Check if user previously dismissed auto-prompts (with consent checking)
     const hasUserDismissedAutoPrompt = useCallback(() => {
         try {
-            return localStorage.getItem('pwa-auto-prompt-dismissed') === 'true';
+            // Check if we have functional consent for storing this preference
+            const consentData = window.localStorage.getItem(
```

Comment:
This bypasses the consent system completely. Should use `useConsentAwareLocalStorage('nextshift_pwa_dismissed', false, 'functional')` instead of direct localStorage access.